Add an "STR#" viewer.

MaddTheSane opened the pull request.

MaddTheSane committed:
Add an "STR#" viewer.

MaddTheSane committed:
Add StringListView to the type controllers list.

MaddTheSane committed:
Minor change.

kainjow commented:
Thanks! I was wanting to do this, but hadn't yet gotten around to it.

I updated the nib style to match the rest of the app, and fixed `where resource.type!.type` to be `where resource.type!.typeString` - wouldn't let me compare OSType to String. Did that compile for you?

Does pascalStringToString() need to check for any besides MacRoman? Doesn't that encoding use all 0-255 characters, so my understanding is it's impossible for something to not be MacRoman. Therefore I don't think we need to check for other encodings. I think GetApplicationTextEncoding() is old school and no longer applies since I don't believe OS X has any type of system-wide encoding anymore.

MaddTheSane commented:
Ah yes, comparing OSType with String… I have a Swift extension that allows me to create OSTypes from a string literal. I guess I forgot this pull request didn't have it.

As for the other encodings: this was a fail-safe I used in my own app: I may have over-engineered it.

Sent from my iPad

> On May 11, 2015, at 9:20 PM, Kevin Wojniak notifications@github.com wrote:
> 
> Thanks! I was wanting to do this, but hadn't yet gotten around to it.
> 
> I updated the nib style to match the rest of the app, and fixed where resource.type!.type to be where resource.type!.typeString - wouldn't let me compare OSType to String. Did that compile for you?
> 
> Does pascalStringToString() need to check for any besides MacRoman? Doesn't that encoding use all 0-255 characters, so my understanding is it's impossible for something to not be MacRoman. Therefore I don't think we need to check for other encodings. I think GetApplicationTextEncoding() is old school and no longer applies since I don't believe OS X has any type of system-wide encoding anymore.
> 
> —
> Reply to this email directly or view it on GitHub.
